pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/osteo/Osteo2GermlinePedConsentVersion3.java

```diff
@@ -0,0 +1,299 @@
+    private static final String DATE_QUESTION_STABLEID_PAR = "ADDENDUM_CONSENT_DOB_PEDIATRIC";
+    private static final String DATE_QUESTION_STABLEID_CHD = "ADDENDUM_CONSENT_PATIENT_DOB_PEDIATRIC";
+    private static final Gson gson = GsonUtil.standardGson();
+
+    private Config dataCfg;
+    private Config varsCfg;
+    private Path cfgPath;
+    private Instant timestamp;
+    private String versionTag;
+    private Config cfg;
+    private ActivityDao activityDao;
+    private SqlHelper sqlHelper;
+    private JdbiVariableSubstitution jdbiVarSubst;
+    private JdbiRevision jdbiRevision;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+
+        this.cfgPath = cfgPath;
+        this.varsCfg = varsCfg;
+
+        String studyGuid = studyCfg.getString("study.guid");
+        if (!studyGuid.equalsIgnoreCase(STUDY)) {
+            throw new DDPException("This task is only for the " + STUDY + " study!");
+        }
+
+        cfg = studyCfg;
+        versionTag = dataCfg.getString("versionTag");
+        timestamp = Instant.now();
+    }
+
+    @Override
+    public void run(Handle handle) {
+        log.info("Executing task: Osteo2GermlinePedConsentVersion3");
+        User adminUser = handle.attach(UserDao.class).findUserByGuid(cfg.getString("adminUser.guid")).get();
+
+        String activityCode = "GERMLINE_CONSENT_ADDENDUM_PEDIATRIC";
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(dataCfg.getString("study.guid"));
+
+        log.info("Changing version of {} to {} with timestamp={}", activityCode, versionTag, timestamp);
+        long ts = this.timestamp.toEpochMilli();
+
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyDto.getGuid(), activityCode, versionTag);
+        RevisionMetadata metaConsent = new RevisionMetadata(ts, adminUser.getId(), reason);
+
+        this.activityDao = handle.attach(ActivityDao.class);
+        this.sqlHelper = handle.attach(SqlHelper.class);
+        this.jdbiVarSubst = handle.attach(JdbiVariableSubstitution.class);
+        this.jdbiRevision = handle.attach(JdbiRevision.class);
+
+        log.info("Revisioning: {} ", activityCode);
+        ActivityVersionDto version3ForConsent = getVersion(handle, studyDto, metaConsent, activityCode, versionTag);
+
+        runGermlinePedConsentUpdate(handle, metaConsent, version3ForConsent, dataCfg);
+
+        //TODO
+        //disable parent/guardian Date Question & add new Content block with sys date
+        disableQuestionAndAddContent(handle, studyDto.getId(), metaConsent, version3ForConsent, DATE_QUESTION_STABLEID_PAR, BLOCK_ADD_PAR);
```

Comment:
Date question was incorrectly code as user input where it should be a system date.
corrected in this revision
disable inout Date question and add as a system date